internal/service/mongo.go

```diff
@@ -0,0 +1,72 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/etf1/kafka-mongo-watcher/internal/mongo"
+	"github.com/gol4ng/logger"
+	mongodriver "go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func (container *Container) GetMongoClient() mongo.Client {
+	if container.mongoClient == nil {
+		var configOptions = container.Cfg.MongoDB.Options
+		var options = []mongo.Option{
+			mongo.WithBatchSize(configOptions.BatchSize),
+			mongo.WithFullDocument(configOptions.FullDocument),
+			mongo.WithMaxAwaitTime(configOptions.MaxAwaitTime),
+		}
+
+		container.mongoClient = mongo.NewClient(
+			container.GetLogger(),
+			options...,
+		)
+	}
+
+	return container.mongoClient
+}
+
+func (container *Container) GetMongoConnection(ctx context.Context) *mongodriver.Database {
```

Comment:
why it need a context to initialize this

Reply:
mongo.Connect() need a context (see below)